TLM/TLM/Manager/Impl/ExtCitizenInstanceManager.cs

```diff
@@ -942,7 +998,7 @@ > ItemClass.Service.Office
                 CitizenInstance.Flags.None;
 
```

Comment:
https://github.com/CitiesSkylinesMods/TMPE/blob/c3137d6d1f252a852aa4097f9e85c9b713322343/TLM/TLM/Manager/Impl/ExtCitizenInstanceManager.cs#L966-L967
replace with
```cs
 if (allowRandomParking && instanceData.m_targetBuilding != 0 && 
    (buildingsBuffer[instanceData.m_targetBuilding].Info &&
     buildingsBuffer[instanceData.m_targetBuilding].Info.m_class.m_service
```
That change will fix possible `NullReferenceException` from #825 (user report)
_I don't think that fixing null ref will solve Stuck cims issue :/_

Reply:
@krzychu124 I've spaced it out a bit but does this look correct:

```cs
            if (extVehicleType == ExtVehicleType.PassengerCar ||
                extVehicleType == ExtVehicleType.Bicycle) {

                if (allowRandomParking
                    && instanceData.m_targetBuilding != 0
                    && buildingsBuffer[instanceData.m_targetBuilding].Info != null
                    && (
                        buildingsBuffer[instanceData.m_targetBuilding].Info.m_class.m_service > ItemClass.Service.Office
                        || (instanceData.m_flags & CitizenInstance.Flags.TargetIsNode) != 0
                       )
                    )
                {

                    randomParking = true;
                }
            }
```

Reply:
The key here is to not comparing `MonoBehaviour` with `null`, because it may return false positive result when object was destroyed in the same frame (not bad, pretty rare situation, we will get one exception only xD ).
More info about null thingies: https://blogs.unity3d.com/2014/05/16/custom-operator-should-we-keep-it/

Reply:
So this:

`buildingsBuffer[instanceData.m_targetBuilding].Info != null`

should be just:

`buildingsBuffer[instanceData.m_targetBuilding].Info`

?

Reply:
Yeah, I know it looks weird, but I think that should cover all cases

Reply:
Updated